File: src/main/java/com/violeth/blockjs/blockjs/CommandInterface.java
package com.violeth.blockjs.blockjs;

import com.violeth.blockjs.blockjs.jsinterface.JSRunner;
import net.kyori.adventure.text.Component;
import org.bukkit.command.CommandExecutor;
import org.jetbrains.annotations.NotNull;

public class CommandInterface implements CommandExecutor {
    @Override
    public boolean onCommand(
        org.bukkit.command.@NotNull CommandSender sender,
        org.bukkit.command.@NotNull Command command,
        @NotNull String label, String[] args
    ) {
        if(args.length > 0) {
            for(var i = 0; i < Commands.list.size(); i++) {
                var commandInst = Commands.list.get(i);
                if(args[0].equalsIgnoreCase(commandInst.key)) {
                    commandInst.handle(sender, args);

                    return true;
                }
            }

            sender.sendMessage(BlockJS.getPluginMessagePrefix().append(
                    Component.text("Unknown command. Type `/blockjs help` for more info")
            ));
        } else {
            sender.sendMessage(BlockJS.getPluginMessagePrefix().append(
                Component.text("Type `/blockjs help` for more info")
            ));
        }

        return true;
    }
}

File: plugin/src/main/kotlin/io/github/blockjs/plugin/CommandInterface.kt
package io.github.blockjs.plugin

import net.kyori.adventure.text.Component
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

class CommandInterface: CommandExecutor {
    override fun onCommand(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<out String>?
    ): Boolean {
        if (args.isNullOrEmpty()) {
            sender.sendMessage(
                Main.pluginMessagePrefix!!.append(
                    Component.text("Type `/blockjs help` for more info")
                )
            )
        } else {
            for (i in 0..< Commands.list.size) {
                val commandInst = Commands.list[i]

                if (args[0].equals(commandInst.key, ignoreCase = true)) {
                    commandInst.handle(sender, args)
                    return true
                }
            }

            sender.sendMessage(
                Main.pluginMessagePrefix!!.append(
                    Component.text("Unknown command. Type `/blockjs help` for more info")
                )
            )
        }

        return true
    }
}
